Public/Src/FrontEnd/Script/Ambients/Transformers/AmbientTransformer.Ipc.cs

```diff
@@ -159,17 +159,23 @@ private bool TryScheduleIpcPip(Context context, ObjectLiteral obj, bool allowUnd
             }
 
             // skip materialization for files
-            FileArtifact[] skipMaterializationFiles = CollectionUtilities.EmptyArray<FileArtifact>();
+            FileOrDirectoryArtifact[] skipMaterializationArtifacts = CollectionUtilities.EmptyArray<FileOrDirectoryArtifact>();
             ArrayLiteral skipMaterializationLiteral = Converter.ExtractArrayLiteral(obj, m_ipcSendLazilyMaterializedDependencies, allowUndefined: true);
             if (skipMaterializationLiteral != null)
             {
-                skipMaterializationFiles = new FileArtifact[skipMaterializationLiteral.Length];
+                skipMaterializationArtifacts = new FileOrDirectoryArtifact[skipMaterializationLiteral.Length];
                 for (int i = 0; i < skipMaterializationLiteral.Length; i++)
                 {
-                    skipMaterializationFiles[i] = Converter.ExpectFile(
+                    Converter.ExpectFileOrStaticDirectory(
                         skipMaterializationLiteral[i],
-                        strict: true,
+                        out var fileArtifact,
+                        out var staticDirectory,
                         context: new ConversionContext(pos: i, objectCtx: skipMaterializationLiteral));
+
+                    // either fileArtifact or staticDirectory is valid at this point (otherwise, ExpectFileOrStaticDirectory would have thrown an exception) 
+                    skipMaterializationArtifacts[i] = fileArtifact.IsValid

```

Comment:
perhaps an assert to ensure we don't get unexpected behavior when this assumption is no longer valid int he future...